### PR TITLE
Add `ConditionalSimplify` and `ConditionalSimplifyDeep` types

### PR DESCRIPTION
**Pull request by skarab42**
Simplifies a type while including and/or excluding certain types from being simplified. 

Related to https://github.com/sindresorhus/type-fest/issues/436#issuecomment-1222084031

### CODE REVIEW

File: source/conditional-simplify.d.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+Simplifies a type while including and/or excluding certain types from being simplified. Useful to improve type hints shown in editors. And also to transform an interface into a type to aide with assignability.
+
+This type is **experimental** and was introduced as a result of this {@link https://github.com/sindresorhus/type-fest/issues/436 issue}. It should be used with caution.
+
+@internal
+@experimental
+@category Object
+*/
+export type ConditionalSimplify<Type, ExcludeType = never, IncludeType = any> = Type extends ExcludeType
```
Comment (sindresorhus) [inline source/conditional-simplify.d.ts:10]: Can we use `unknown` instead of `any`?

File: source/conditional-simplify.d.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
+Simplifies a type while including and/or excluding certain types from being simplified. Useful to improve type hints shown in editors. And also to transform an interface into a type to aide with assignability.
+
+This type is **experimental** and was introduced as a result of this {@link https://github.com/sindresorhus/type-fest/issues/436 issue}. It should be used with caution.
+
```
Comment (sindresorhus) [inline source/conditional-simplify.d.ts:5]: Can you add a `@see` for the `Simplify` type?

**Comment by sindresorhus** [comment]
@skarab42 You have done a lot of great work here lately. Would you be interested in becoming a maintainer? No worries if not :)

**Comment by skarab42** [comment]
> @skarab42 You have done a lot of great work here lately. Would you be interested in becoming a maintainer? No worries if not :)

@sindresorhus Thanks, it's a pleasure and I'm learning a lot :) I would be honoured to become a maintainer 💜

**Comment by sindresorhus** [comment]
Awesome! Welcome :)

**Comment by sindresorhus** [comment]
Even as a maintainer, I still prefer to continue the flow with using pull requests for type additions and larger changes.

**Comment by skarab42** [comment]
> Even as a maintainer, I still prefer to continue the flow with using pull requests for type additions and larger changes.

It's a flow that suits me well :) let's keep it that way 🚀

**Pull request merged**